libgadget/forcetree.c

```diff
@@ -515,7 +515,7 @@ force_tree_build_single(const struct TreeBuilder tb, const int npart)
  */
 void force_create_node_for_topnode(int no, int topnode, unsigned int bits, unsigned int x, unsigned int y, unsigned int z, int *nextfree, const int lastnode)
 {
-    unsigned int i, j, k;
+    int i, j, k;
```

Comment:
Why is it important to keep these signed?

Reply:
Look at line 537: 2*i - 1, which is -1 if i == 0. if i is unsigned that comes out as 2**32 - 1. I am worried about this pull for that reason: there is the potential to introduce subtle bugs worse than the compiler bug we are working around. Please review carefully.